sample_apps/rso_sample_apps/rust/src/config.rs

```diff
@@ -0,0 +1,330 @@
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub(crate) struct Server {
+    /// The server address
+    pub host: String,
+    // The port
+    pub port: u16,
+    /// The server TLS configuration
+    pub tls: Option<Tls>,
+}
+
+impl Server {
+    pub fn addr(&self) -> String {
+        format!("{}:{}", self.host, self.port)
+    }
+}
+
+/// Configuration holds the  required parameters to connect to a Git provider.
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+pub(crate) struct Configuration {
+    /// The  client ID.
+    pub client_id: String,
+    /// The client secret.
+    pub client_secret: String,
+    /// The provider URL.
+    pub provider_url: String,
+    /// The server configuration.
+    pub server: Server,
+    /// The callback host.
+    pub callback_host: String,
+    /// The RiotGamesAPI Token
+    pub api_token: String,
+    /// The account data URL
+    pub account_data_url: String,
+    /// The champion data URL
+    pub champion_data_url: String,
+}
+
+impl Configuration {
+    /// The callback URL that the OAuth provider will redirect to after authentication.
+    pub fn callback_url(&self) -> String {
+        let protocol = match self.server.tls {
+            Some(_) => "https://",
+            None => "http://",
+        };
+        format!("{}{}/oauth", protocol, self.callback_host)
+    }
+
+    /// Returns the provide token endpoint.
+    pub fn token_url(&self) -> String {
+        format!("{}/token", self.provider_url)
+    }
+
+    /// Returns the URL of the OAuth authorize endpoint.
+    ///
+    ///  The authorize endpoint is where the user is redirected to grant the application permission to
+    /// access their data. The user will be prompted to log in to their account  and grant the application
+    /// the requested permissions. Once the user has granted the permissions, they will be redirected
+    /// back to the application with an authorization code.
+    pub fn authorize_url(&self) -> String {
+        format!("{}/authorize", self.provider_url)
+    }
+
+    /// Returns the  sign in URL which opens the browser authenticates, and redirects to the
+    /// callback URL to complete authentication.
+    pub fn sign_in_url(&self) -> String {
+        format!(
+            "{}?redirect_uri={}&client_id={}&response_type=code&scope=openid",
+            self.authorize_url(),
+            self.callback_url(),
+            self.client_id,
+        )
+    }
+}
+
+/// Parse a YAML  file into a `Configuration` object.
+///
+/// # Arguments
+///
+/// * `filepath`: The path to the YAML file.
+///
+/// # Returns
+///
+/// A `Result` containing the `Configuration` object or an error message.
+pub(crate) fn parse_file(filepath: String) -> Result<Configuration, String> {
+    let contents = read_to_string(filepath).map_err(|err| format!("error reading file - {err}"))?;
+    serde_yaml::from_str(&contents).map_err(|err| format!("error parsing contents - {err}"))
```

Comment:
The `config` crate [does this for you in a standardized way](https://github.com/mehcode/config-rs/blob/master/examples/simple/main.rs).